aten/src/ATen/native/README.md

```diff
@@ -476,6 +476,21 @@ as `Tensor &`, which 1) allowed changing which `TensorImpl` the `Tensor` itself
 was not necessary to allow the underlying data to change. (This was like using `T * const` when we
 wanted `const T*`.)
 
+### `autogen`
+
+```
+- func: my_op_(Tensor(a!) self) -> Tensor(a!)
+...
+  autogen: my_op, my_op.out
+```
+
+For an in-place variant of a native function (op name ends with an `_`), we offer the option to
```

Comment:
I feel like there are a couple different ideas here that would benefit from splitting into multiple paragraphs?
- This is used to specify which function the codegen should generate implementations for
- This can be used by
  - inplace function to get out-of-place and out= variants
  - functional function to get out= variant
- This doesn't support view, manual binding and composite ops
- The generated kernels are trivial implementation that do copies, they can be found at...

Reply:
+1  ^
Is the recommendation that any new op added, if it conforms to the above restrictions, should add the autogen key in order to better enable functionalization?

Reply:
Yeah I think it’s recommended. For functionalization I think the autogenerate feature on future inplace ops is useful so I think we should recommend it.

Reply:
+1 to Alban's feedback.

Also, I think it's worth mentioning that it's not "optional": All ops that follow the criteria you mentioned are now required to have functional + out= variants, and the codegen will automatically try to create them if they're missing. The main point of the `autogen` keyword is to be more explicit: every native function in ATen can still be found in native_functions.yaml (the codegen isn't "secretly" generating anything new without any indication from the yaml)